__tests__/components/TransactionList/TransactionRow.spec.ts

```diff
@@ -0,0 +1,888 @@
+        });
+    });
+
+    describe('fetchTransaction', () => {
+        const runBasicFetchTransactionTests = async (transaction, expectedResult) => {
+            // Arrange:
+            const wrapper = getTransactionRowWrapper({ currentProfile: testnetProfile, currentAccount: WalletsModel1 }, { transaction });
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // @ts-ignore - for private method
+            jest.spyOn(vm, 'hasMissSignatures', 'get').mockReturnValue(true);
+            jest.spyOn(vm, 'needsCosignature');
+
+            // Act:
+            // @ts-ignore - for private method
+            await vm.fetchTransaction();
+
+            // Assert:
+            // @ts-ignore - for private method
+            expect(vm.transactionSigningFlag).toBe(expectedResult);
+            expect(vm.needsCosignature).not.toBeCalled();
+        };
+
+        test('needsCosignature called when transaction status confirmed', async () => {
+            // Arrange:
+            const aggregateTransaction = mockAggregateTransaction();
+
+            dispatch.mockImplementation((type: string) => {
+                if (type === 'transaction/FETCH_TRANSACTION_STATUS') {
+                    return Promise.resolve(new TransactionStatus(TransactionGroupEnum.Confirmed, '1', mockDeadline));
+                } else if (type === 'transaction/LOAD_TRANSACTION_DETAILS') {
+                    return Promise.resolve(aggregateTransaction);
+                }
+            });
+
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile, currentAccount: WalletsModel1 },
+                { transaction: aggregateTransaction },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // @ts-ignore - for private method
+            jest.spyOn(vm, 'hasMissSignatures', 'get').mockReturnValue(true);
+            jest.spyOn(vm, 'needsCosignature');
+
+            // Act:
+            // @ts-ignore - for private method
+            await vm.fetchTransaction();
+
+            // Assert:
+            // @ts-ignore - for private method
+            expect(vm.needsCosignature).toHaveBeenCalled();
+        });
+
+        test('needsCosignature not called  when transaction status failed', async () => {
+            // Arrange:
+            const aggregateTransaction = mockAggregateTransaction();
+
+            dispatch.mockImplementation((type: string) => {
+                if (type === 'transaction/FETCH_TRANSACTION_STATUS') {
+                    return Promise.resolve(new TransactionStatus(TransactionGroupEnum.Failed, '1', mockDeadline));
+                } else if (type === 'transaction/LOAD_TRANSACTION_DETAILS') {
+                    return Promise.resolve(aggregateTransaction);
+                }
+            });
+
+            await runBasicFetchTransactionTests(aggregateTransaction, true);
```

Comment:
why is expected results true in these two failure cases?

Reply:
Actually, `transactionSigningFlag` will be always true, when `fetchTransaction` method is called.

But `needsCosignature` method is the key to updating `transactionSigningFlag` 

So I beieve it's pointless to include `vm.transactionSigningFlag` in this case